src/Field.tsx

```diff
@@ -113,19 +119,58 @@ export class Field<Props extends FieldAttributes = any> extends React.Component<
     );
   }
 
+  handleChange = (e: React.ChangeEvent<any>) => {
+    const { handleChange, validateOnChange } = this.context.formik;
+    handleChange(e); // Call Formik's handleChange no matter what
+    if (!!validateOnChange && !!this.props.validate) {
+      this.runFieldValidations(e.target.value);
+    }
+  };
+
+  handleBlur = (e: any) => {
+    const { handleBlur, validateOnBlur } = this.context.formik;
+    handleBlur(e); // Call Formik's handleBlur no matter what
+    if (validateOnBlur && this.props.validate) {
+      this.runFieldValidations(e.target.value);
+    }
+  };
+
+  runFieldValidations = (value: any) => {
+    const { setFieldError } = this.context.formik;
+    const { name, validate } = this.props;
+    // Call validate fn
+    const maybePromise = (validate as any)(value);
+    // Check if validate it returns a Promise
+    if (isPromise(maybePromise)) {
+      (maybePromise as Promise<any>).then(
+        () => setFieldError(name, undefined),
+        error => setFieldError(name, error)
+      );
+    } else {
+      // Otherwise set the error
+      setFieldError(maybePromise);
```

Comment:
Shouldn't this be `setFieldError(name, maybePromise)`? I think that this is causing an error for me...

Reply:
Yeah, adding `name` here fixes this for me. The debugger also told me that `setDeep` was being called with my error message as the path so this seems to be the culprit. I opened #290 to fix it